test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -1037,71 +1050,92 @@ public async Task DiagnosticSourceExceptionCallBackIsNotReceivedForExceptionsHan
                 })
                 .Build();
 
-        var builder = WebApplication.CreateBuilder();
-        builder.Logging.ClearProviders();
-        var app = builder.Build();
+            services.AddOpenTelemetry()
+                .WithTracing(builder => builder
+                    .AddInstrumentation(() => this.tracerProvider));
+        }
 
-        app.UseExceptionHandler(handler =>
-        {
-            handler.Run(async (ctx) =>
+        using (var client = this.factory
+            .WithWebHostBuilder(builder =>
             {
-                await ctx.Response.WriteAsync("handled").ConfigureAwait(false);
-            });
-        });
-
-        app.Map("/error", ThrowException);
-
-        static void ThrowException(IApplicationBuilder app)
+                builder.ConfigureTestServices(ConfigureTestServices);
+                builder.ConfigureLogging(loggingBuilder => loggingBuilder.ClearProviders());
+                builder.Configure(app => app
+                    .UseExceptionHandler(handler =>
+                    {
+                        handler.Run(async (ctx) =>
+                        {
+                            await ctx.Response.WriteAsync("handled").ConfigureAwait(false);
+                        });
+                    }));
+            })
+            .CreateClient())
         {
-            app.Run(context =>
+            try
             {
-                throw new Exception("CustomException");
-            });
-        }
-
-        _ = app.RunAsync();
+                using var request = new HttpRequestMessage(HttpMethod.Get, "/api/error");
 
-        using var client = new HttpClient();
-        try
-        {
-            await client.GetStringAsync("http://localhost:5000/error").ConfigureAwait(false);
-        }
-        catch
-        {
-            // ignore 500 error.
+                // Act
+                using var response = await client.SendAsync(request).ConfigureAwait(false);
+            }
+            catch
+            {
+                // ignore exception
+            }
         }
 
         Assert.Equal(0, numberOfExceptionCallbacks);
         Assert.Equal(0, numberOfUnSubscribedEvents);
-        Assert.Equal(2, numberofSubscribedEvents);
-
-        await app.DisposeAsync().ConfigureAwait(false);
+        Assert.Equal(2, numberOfSubscribedEvents);
     }
 
     [Fact]
     public async Task RouteInformationIsNotAddedToRequestsOutsideOfMVC()
```

Comment:
FYI - this test is being removed in https://github.com/open-telemetry/opentelemetry-dotnet/pull/5026

Reply:
Ok, I didn't change it further now.